File: clojure-pid-service/src/org/cyverse/dataone/ClojurePidServiceFactory.clj
(ns org.cyverse.dataone.ClojurePidServiceFactory
  (:gen-class :implements [org.irods.jargon.pid.pidservice.DataOnePidServiceFactory]))

(defn -instance [this publication-context irods-account]
  (org.cyverse.dataone.ClojurePidService. irods-account publication-context))

File: clojure-pid-service/src/java/org/cyverse/dataone/ClojurePidServiceFactory.java
package org.cyverse.dataone;

import org.irods.jargon.core.connection.IRODSAccount;
import org.irods.jargon.dataone.pidservice.AbstractDataOnePidFactory;
import org.irods.jargon.dataone.pidservice.AbstractDataOnePidServiceAO;
import org.irods.jargon.dataone.plugin.PublicationContext;

public class ClojurePidServiceFactory extends AbstractDataOnePidFactory {

    public AbstractDataOnePidServiceAO instance(PublicationContext publicationContext, IRODSAccount irodsAccount) {
        return new ClojurePidService(irodsAccount, publicationContext);
    }
}
